library-viewbinding/src/main/java/com/xwray/groupie/viewbinding/BindableItem.java

```diff
@@ -0,0 +1,97 @@
+package com.xwray.groupie.viewbinding;
+
+import androidx.annotation.CallSuper;
+import androidx.annotation.NonNull;
+import androidx.databinding.ViewDataBinding;
+import androidx.viewbinding.ViewBinding;
+
+import android.view.View;
+
+import com.xwray.groupie.Item;
+import com.xwray.groupie.OnItemClickListener;
+import com.xwray.groupie.OnItemLongClickListener;
+
+import java.util.List;
+
+/**
+ * The base unit of content for a GroupAdapter.
+ * <p>
+ * Because an Item is a Group of size one, you don't need to use Groups directly if you don't want;
+ * simply mix and match Items and add directly to the adapter.
+ * <p>
+ * If you want to use Groups, because Item extends Group, you can mix and match adding Items and
+ * other Groups directly to the adapter.
+ *
+ * @param <T> The ViewBinding subclass associated with this Item.
+ */
+public abstract class BindableItem<T extends ViewBinding> extends Item<GroupieViewHolder<T>> {
+
+    public BindableItem() {
+        super();
+    }
+
+    protected BindableItem(long id) {
+        super(id);
+    }
+
+    @NonNull
+    protected abstract T initializeViewBinding(@NonNull View view);
+
+    @NonNull
+    @Override
+    public GroupieViewHolder<T> createViewHolder(@NonNull View itemView) {
+        T binding = initializeViewBinding(itemView);
+        return new GroupieViewHolder<>(binding);
+    }
+
+    @Override
+    public void bind(@NonNull GroupieViewHolder<T> viewHolder, int position) {
+        throw new RuntimeException("Doesn't get called");
+    }
+
+    @Override
+    public void bind(@NonNull GroupieViewHolder<T> viewHolder, int position, @NonNull List<Object> payloads) {
+        bind(viewHolder.binding, position, payloads);
+    }
+
+    /**
+     * Perform any actions required to set up the view for display.
+     *
+     * @param viewHolder          The viewHolder to bind
+     * @param position            The adapter position
+     * @param payloads            Any payloads (this list may be empty)
+     * @param onItemClickListener An optional adapter-level click listener
+     * @param onItemLongClickListener An optional adapter-level long click listener
+     */
+    @CallSuper
+    @Override
+    public void bind(@NonNull GroupieViewHolder<T> viewHolder, int position, @NonNull List<Object> payloads, OnItemClickListener onItemClickListener, OnItemLongClickListener onItemLongClickListener) {
+        super.bind(viewHolder, position, payloads, onItemClickListener, onItemLongClickListener);
+        T binding = viewHolder.binding;
+        if (binding instanceof ViewDataBinding) {
+            ((ViewDataBinding) binding).executePendingBindings();
+        }
```

Comment:
This will certainly help users who want to use both DataBinding and ViewBinding.
The only downside I see here is, even though the user wants to only use ViewBinding, he will still have to add data-binding as a dependency for this code to even compile.
What are your thoughts?

Reply:
Oh, I didn't check that case. Thank you for the pointing :)
Hmm... how about changing dependency declaration for `databinding-runtime` and `databinding-common` from `compileOnly` to `implementation`? This allows the user to use groupie-viewbinding artifact without enabling DataBinding in their build.gradle.

Reply:
Changing to `implementation` could solve the compile issue.
But this still adds two extra dependencies `databinding-runtime`, `databinding-common` (for users only using ViewBinding), just because we library authors chose to implement it this way.
And no one loves extra dependency, It will come up as another issue later:

> Why the heck did I get DataBinding when I add `library-viewbinding` to my project?

Reply:
I'm sorry for late reply.

Indeed the dependencies to data binding libraries are redundant for the users only using View Binding, but these are necessary for this library to care about all possible use cases for the safety.
As [I commented above](https://github.com/lisawray/groupie/pull/325#issuecomment-589940382), `<T extends ViewBinding>` accepts the classes generated by Data Binding, which are subclasses of `ViewDataBinding`, because [`ViewDataBinding` implements `ViewBinding` since 3.6.0](https://androidx.tech/artifacts/databinding/databinding-runtime/3.6.0-source/androidx/databinding/ViewDataBinding.java.html). This means even if we don't intend to support Data Binding in `groupie-viewbinding` and don't add the implementation pointed in this comment thread, users CAN use this library with Data Binding, and sometimes it doesn't work (for detail, see the pages listing below).

- https://stackoverflow.com/questions/53043412/android-why-use-executependingbindings-in-recyclerview
- https://medium.com/androiddevelopers/android-data-binding-recyclerview-db7c40d9f0e4
- https://developer.android.com/topic/libraries/data-binding/generated-binding

I prefer that the APIs of libraries are designed to prevent wrong usage because API definition is the only way to force its usage, and it is impossible to implement to exclude `ViewDataBinding` and its subclasses in compile time when considering the implementers of `ViewBinding`, so I added the consideration for Data Binding in the new library to support Data Binding as a special variation of View Binding.

Reply:
I rather we do not force databinding on users who want to use viewbinding. I appreciate you wanting to code the library defensively but I think it's more accurate to have most users of this library not get an extra dependency just because *some* users might need it.

Reply:
@nashcft I still think this should be removed in order to merge this PR.